### DISCUSSION TITLE
Trying to run Machinaris

### DISCUSSION THREAD

**Discussion by JoshRedman90** [guydavis/machinaris#760] [Q&A] [answered]
I am currently trying to run Machinaris on a Windows 10 based machine.

When I run Machinaris and check on the page "blockchains" the error provided is as follows: https://hastebin.com/emawenocim.apache

Hopefully someone can help me figure this out. Thanks.

**Comment by JoshRedman90** [comment]
I believe the issue was that my router wasn't port forwarded properly. Seems to be working now.

**Reply by JoshRedman90** [reply]
Nevermind, this is seemingly still an issue: https://hastebin.com/iborihiquh.apache

**Comment by guydavis** [comment, marked as answer]
> I am currently trying to run Machinaris on a Windows 10 based machine.
> When I run Machinaris and check on the page "blockchains" the error provided is as follows: https://hastebin.com/emawenocim.apache

Hi, sorry to hear that.  The error snippet you show is likely from the Chia debug.log for the blockchain itself.  If you are receving an Internal Server Error from the Machinaris WebUI for just the Blockchains page, please [troubleshoot the web log](https://github.com/guydavis/machinaris/wiki/Troubleshooting#internal-server-error) instead.

**Reply by JoshRedman90** [reply]
While viewing the web log this is what it says: https://hastebin.com/ataragapek.yaml

**Reply by guydavis** [reply]
```
2022-09-03 22:47:00 [1198] [INFO] 172.18.0.1 - - [03/Sep/2022:22:47:00 -0400] "GET /blockchains HTTP/1.1" 200 25424 "http://192.168.2.85:8926/blockchains" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36"
```

The blockchains page returns a HTTP 200 so the "error" must be something that you are seing on the page that loads as there is no Internal Server Error.  Would you mind sharing the user-visible error you are referring to above?  A screenshot would be best.

**Reply by JoshRedman90** [reply]
![brave_Zgrnr3c6io](https://user-images.githubusercontent.com/111439576/188295066-736fcdc9-f17f-4b12-9411-08370e12bc3f.png)

**Reply by guydavis** [reply]
Very interesting, a picture is worth a thousand words.  That means that the command `chia show -s` is itself returning that error as well.  I would recommend you [reset your blockchain config](https://github.com/guydavis/machinaris/wiki/Troubleshooting#reset-blockchain-config) - creates a fresh config.  Wait at least 30 minutes after restarting the container to see if the Chia blockchain services are able to get running cleanly again.  Hope this helps!

**Reply by JoshRedman90** [reply]
Ok thank you, that seems to be working for now. I was also wondering if you could tell me the correct format for a network share. 

![image](https://user-images.githubusercontent.com/111439576/188296096-cb1866dc-e47d-4a3a-8a0e-1dab717c31a7.png)

As you can see here this is how I was entering the information, but here is the actual location

![image](https://user-images.githubusercontent.com/111439576/188296137-848fc7f4-4179-47e1-bb37-f843d29bd1e8.png)

**Reply by JoshRedman90** [reply]
The plots are also located right on the DriveX folder

**Reply by guydavis** [reply]
Hi, so there are two ways you can go to [mount remote Windows file shares](https://github.com/guydavis/machinaris/wiki/Windows#network-share-access) into the Docker container.  In your case above, you could put the '/DrivesX' suffix onto each host path.

- //192.168.2.65/Disks/Drive1 -> /Drive1
- //192.168.2.65/Disks/Drive2 -> /Drive2

This will map the correct unique host path to a unique in-container path.  The error you were likely getting was that you were trying to mount the same host path (`//192.168.2.65/Disks`) repeatedly.

A simpler alternative, is to mount just the parent path one time:
- //192.168.2.65/Disks  -> /Disks

This *should* mount the entire parent folder, containing all your drives folders, inside the container at `/Disks`.   Then in future, if you add another `/Disks/DriveX` folder it *should* automatically appear, not requiring a new volume mounting line.  I say *should* as I am not 100% that drive shortcuts like that will translate inside the container... if they don't then you'll need to do the more tedious approach of one line per disk as corrected above.

To check, try just the single volume mounting line with `/Disks` in your `docker-compose.yml`.  As most blockchains still require a direct path to each plots folder, be sure to manually edit your `docker-compose.yml` to have a `plots_dir` environment variable like:
```
/Disks/Drive1:/Disks/Drive2:/Disks/Drive3
```
and so on.  While there may be a single mounted folder (`/Disks`), the blockchain still needs to look in each child folder for plots, which is the purpose of the `plots_dir` environment variable in the `docker-compose.yml`.

To test this setup, launch the container with this updated `docker-compose.yml`, then from Docker Desktop (or similar) get into a terminal shell within the container itself.  Then type:
```
cd /Disks
ls -1 Drive1/*.plot | wc -l
```
This should return a count of the plot files on Drive1.  If you get an error or zero, then you may need to use the multiple volume mounting line (one per Drive) as I listed at start of my response.  Hope this helps!

**Reply by JoshRedman90** [reply]
Alright, I've used both this, 
![image](https://user-images.githubusercontent.com/111439576/188332174-17db103a-5aaa-47e9-abe9-c5bb3b17a809.png)
and this,
![image](https://user-images.githubusercontent.com/111439576/188332193-0cf81b46-2b85-4113-85ae-a9848ceccee3.png)

I'm not so sure I am understanding properly, neither are able to use the command you provided.

**Reply by guydavis** [reply]
Thanks for sharing your `docker-compose.yml`. You are using local Docker Volume mounts (T:/, X:/, etc) but providing a Remote Share path (//192.168.2.85).  Please ensure you are setting the Location to Remote on [Machinaris Setup](https://www.machinaris.app) site.

![image](https://user-images.githubusercontent.com/12936/188334488-80a77d2f-d389-4c5f-8171-a2c5f99e8ae0.png)

You can take the generated config, changing only `plots_dir` and use that.  Hope this helps!

**Reply by JoshRedman90** [reply]
Ah, thats weird as when I use the machinaris.app webpage it shows as this

![image](https://user-images.githubusercontent.com/111439576/188337347-e1174a45-0220-4b48-91a4-808614aa8ee9.png)

I didnt realize that should have been set to remote. Your output looks a little different to mine.

**Reply by JoshRedman90** [reply]
It only seems to give me the remote option on the second option.

**Reply by JoshRedman90** [reply]
But I assume you meant something like this?
![image](https://user-images.githubusercontent.com/111439576/188338062-91dbeb42-5577-468f-bd96-c367f84e2e6d.png)

**Reply by guydavis** [reply]
> Ah, thats weird as when I use the machinaris.app webpage it shows as this
> 
> ![image](https://user-images.githubusercontent.com/111439576/188337347-e1174a45-0220-4b48-91a4-808614aa8ee9.png)
> 
> I didnt realize that should have been set to remote. Your output looks a little different to mine.

Yes, please use the Plus/Minus icon buttons to remove the duplicated row in your screenshot.  Please note that my screenshot had a single row for Plots. You only need one for Plots in your case.  You can add more, including to temporary storage for plotting. If you don't see the options, then refresh the page.
